recipe/recipe.yaml

```
@@ -0,0 +1,228 @@
+
+schema_version: 1
+
+context:
+  version: 2.3.1
+  # -1 means "not a dev version", a number > -1 will emit postfix "rc${{ rc_num }}"
+  rc_num: "-1"
+  # numpy will by default use the ABI feature level for the first numpy version
+  # that added support for the oldest currently-supported CPython version; see
+  # https://github.com/numpy/numpy/blob/v2.0.0rc1/numpy/_core/include/numpy/numpyconfig.h#L124
+  default_abi_level: 1.23
+  # ensure is_freethreading gets detected as a used variable
+  touch_is_freethreading: ${{ is_freethreading }}
+
+package:
+  name: numpy
+  version: ${{ version }}${{ "rc" + rc_num if rc_num != "-1" else "" }}
```

I'd like the precedence clearer here. Also, string concatenation in jinja should use `~` AFAIU; `+` is unreliable.
```suggestion
  version: ${{ version }}${{ ("rc" ~ rc_num) if rc_num != "-1" else "" }}
```

---

done, using an int for `rc_num`.